codypy/agent.py

```diff
@@ -40,61 +37,38 @@ async def initialize_agent(
         """
 
         async def _handle_response(response: Any) -> None:
+            # TODO: Consider attaching CodyAgentSpecs to CodyAgent
```

Comment:
Already done in init as `self.agent_specs = agent_specs`.
So, no need to have a new instance below. The line 41 can be removed then.

Reply:
I've re-worked this slightly to include both the AgentSpecs and the CodyAgentSpecs in the debug log. You're right the initial code was not clear about this.

Reply:
Rebased after your last commit